src/celengine/skygrid.cpp

```diff
@@ -260,20 +263,20 @@ SkyGrid::longitudeLabel(int longitude, int longitudeStep) const
     if (m_longitudeDirection == IncreasingClockwise)
         longitude = (totalUnits - longitude) % totalUnits;
 
-    out << longitude / baseUnit << baseUnitSymbol;
-    if (longitudeStep % baseUnit != 0)
-    {
-        out << ' ' << setw(2) << setfill('0') << (longitude / MIN) % 60 << minuteSymbol;
-        if (longitudeStep % MIN != 0)
-        {
-            out << ' ' << setw(2) << setfill('0') << (longitude / SEC) % 60;
-            if (longitudeStep % SEC != 0)
-                out << '.' << setw(3) << setfill('0') << longitude % SEC;
-            out << secondSymbol;
-        }
-    }
+    std::string result = fmt::format("{}{}", longitude / baseUnit, baseUnitSymbol);
+    if (longitudeStep % baseUnit == 0)
+        return result;
+
+    auto it = fmt::format_to(std::back_inserter(result), " {:02}{}", (longitude / MIN) % 60, minuteSymbol);
+    if (longitudeStep % MIN == 0)
+        return result;
+
+    it = fmt::format_to(it, " {:02}", (longitude / SEC) % 60);
```

Comment:
maybe better use `std::back_inserter(result)` as well? so the code is easier to understand and less dependent on the previous lines

Reply:
Ok, that's how I did it originally...